src/polymer/elements/paper-item-v1-to-v2.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import * as dom5 from 'dom5';
+import stripIndent = require('strip-indent');
+
+import {registry} from '../../registry';
+import {HtmlRule} from '../../html/rule';
+import {ParsedHtmlDocument, Severity, Warning} from 'polymer-analyzer';
+import {FixableWarning} from '../../warning';
+
+class PaperItemV1ToV2 extends HtmlRule {
+  code = 'paper-item-v1-to-v2';
+  description = stripIndent(`
+    ADD A DESCRIPTION
+  `).trim();
+
+  async checkDocument(parsedDocument: ParsedHtmlDocument) {
+    const warnings: Warning[] = [];
+
+    const paperIconItems = dom5.nodeWalkAll(
+      parsedDocument.ast,
+      dom5.predicates.hasTagName('paper-icon-item'),
+      undefined,
+      dom5.childNodesIncludeTemplate
+    );
+
+    const checkNode = (node: dom5.Node) => {
+      if (node.tagName !== undefined &&
+        !dom5.hasAttribute(node, 'slot') &&
+        dom5.hasAttribute(node, 'item-icon')
+      ) {
+        const startTagSourceRange =
+          parsedDocument.sourceRangeForStartTag(node)!;
+        const warning = new FixableWarning({
+          parsedDocument,
+          code: this.code,
+          message: 'ADD MESSAGE',
+          severity: Severity.WARNING,
+          sourceRange: startTagSourceRange
+        });
+
+        const [startOffset, endOffset]
+          = parsedDocument.sourceRangeToOffsets(startTagSourceRange);
+        const startTagText =
+          parsedDocument.contents.slice(startOffset, endOffset);
+        const isSelfClosing = startTagText.endsWith('/>');
+
+        warning.fix = [{
+          range: startTagSourceRange,
+          replacementText:
+            startTagText.slice(0, isSelfClosing ? -2 : -1) +
+            ` slot="item-icon"` +
+            (isSelfClosing ? '/' : '') + '>',
+        }];
+
+        warnings.push(warning);
+      }
+    };
+
+    for (const paperIconItem of paperIconItems) {
+      let suspectNodes = Array.from(paperIconItem.childNodes!);
+
+      const nodeIsTemplateExtension = (node: dom5.Node) => {
```

Comment:
Consider pulling this function out into `src/html/util.ts`

Reply:
I'll do this in #118 and then use it here.